SingularityExecutor/src/main/java/com/hubspot/singularity/executor/SingularityExecutorMonitor.java

```diff
@@ -328,6 +351,52 @@ private void onSuccessThrows(ProcessBuilder processBuilder) {
           wasKilled = task.wasKilled();
 
           if (!wasKilled) {
+
+            SingularityTaskId taskId = SingularityTaskId.valueOf(task.getTaskDefinition().getTaskId());
+            Optional<SingularityRequestWithState> request = requestManager.getRequest(taskId.getRequestId());
+
+            HealthcheckOptions options = deployManager.getDeploy(taskId.getRequestId(), taskId.getDeployId()).get().getHealthcheck().get();
+
+            Integer healthcheckMaxRetries = options.getMaxRetries().or(configuration.getHealthcheckMaxRetries());
+
+            Optional<String> expectedHealthCheckResultFilePath = task.getTaskDefinition().getHealthCheckResultFilePath();
+            if (expectedHealthCheckResultFilePath.isPresent()) {
+              long healthCheckStart = System.currentTimeMillis();
+              try {
+                Retryer<String> retryer = RetryerBuilder.<String>newBuilder()
+                    .retryIfResult(path -> !new File(path).exists())
+                    .withWaitStrategy(WaitStrategies.fixedWait(1L, TimeUnit.SECONDS))
+                    .withStopStrategy(StopStrategies.stopAfterAttempt(healthcheckMaxRetries))
+                    .build();
+
+                retryer.call(() -> expectedHealthCheckResultFilePath.get());
+                long healthCheckEnd = System.currentTimeMillis();
+
+                taskManager.saveHealthcheckResult(new SingularityTaskHealthcheckResult(
+                    Optional.of(200),
+                    Optional.of(healthCheckEnd - healthCheckStart),
+                    healthCheckEnd,
+                    Optional.of("Health check success"),
+                    Optional.absent(),
+                    taskId,
+                    Optional.of(true)));
+
+              } catch (ExecutionException | RetryException e) {
+                finishTask(task, TaskState.TASK_KILLED, "Task timed out on health checks.", Optional.<String> absent());
+
+                long healthCheckEnd = System.currentTimeMillis();
+                taskManager.saveHealthcheckResult(new SingularityTaskHealthcheckResult(
+                    Optional.of(408),
+                    Optional.of(healthCheckEnd - healthCheckStart),
+                    healthCheckEnd,
+                    Optional.absent(),
+                    Optional.of("Task timed out on health checks."),
+                    taskId,
+                    Optional.of(true)));
+                return;
+              }
+            }
+
             processRunningTasks.put(task.getTaskId(), submitProcessMonitor(task, processBuilder));
```

Comment:
The task process doesn't actually start running until after this line. We'd want to make sure the process + monitors are up and running before we start looking at health checks. Ideally it is after we start and right before the TASK_RUNNING. Something like [here](https://github.com/HubSpot/Singularity/blob/master/SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/SingularityExecutorTaskProcessCallable.java#L27) I think would make it much simpler.